### PR TITLE
Refactor UserTransformers

### LINKED ISSUE #419: Cleanup user transformers
https://github.com/pelican-dev/panel/blob/main/app/Transformers/Api/Application/UserTransformer.php

https://github.com/pelican-dev/panel/blob/main/app/Transformers/Api/Client/UserTransformer.php
https://github.com/pelican-dev/panel/blob/main/app/Transformers/Api/Client/AccountTransformer.php

### PR DESCRIPTION
**Pull request by Boy132**
Closes #419

Removes the unnecessary `AccountTransformer` and updates the client api `UserTransformer` to match the application api `UserTransformer`. Also renames `toVueObject` to `toReactObject`,

### CODE REVIEW

**Comment by Boy132** [comment]
@TekExplorer Let me know what you think!

**Review by TekExplorer — COMMENTED** [review]
Things this is going to affect:
- Account
  - `id` was removed
    - This is not a _bad_ thing - clients dont need to be aware of database ids
    - Breaking change: anything depending on the id being present in the api will break.
    - I'm not sure what the best solution to this is. Document the transition to `uuid`?
  - `admin` was renamed to `root_admin`
    - Breaking change. Many more api consumers are likely to be impacted by this change
  - `created_at`, `updated_at`, `uuid`, and `2fa` (or, `totp_enabled`) are introduced
- Subusers _and_ Activity Log `includes: actor`
  - `2fa_enabled` was renamed to `2fa` 
    - I dont like this. If you intend to rename it, call it `totp_enabled`, which is more descriptive
     - Breaking change
  - `image` was removed
    - Technically a breaking change, but... who's using this?
  - `first_name`, `last_name`, and `updated_at` are introduced

The following files need to be updated, at minimum:
[resources/scripts/api/definitions/user/transformers.ts](https://github.com/pelican-dev/panel/blob/main/resources/scripts/api/definitions/user/transformers.ts#L21)
[resources/scripts/api/server/users/getServerSubusers.ts](https://github.com/pelican-dev/panel/blob/main/resources/scripts/api/server/users/getServerSubusers.ts#L9)
-- The front end uses `toVueObject` (`toReactObject`) so it's not affected by `account`

I would also recommend adding the new fields to the appropriate objects so that future updates to the panel will be aware of the extra information (though, I understand the intent is to replace the frontend again?)

That being said, I do think it makes a ton of sense to consolidate the two together.
All we need to consider now is API compatibility.

File: app/Transformers/Api/Client/UserTransformer.php [hunk 1]
```diff
@@ -19,15 +18,19 @@ public function getResourceName(): string
      * Transforms a User model into a representation that can be shown to regular
      * users of the API.
      */
-    public function transform(User $model): array
+    public function transform(User $user): array
     {
         return [
```
[the remark this answers is not in the document]
Comment (Boy132) [reply app/Transformers/Api/Client/UserTransformer.php:24]: I think we should just document the removal.

File: tests/Integration/Api/Client/AccountControllerTest.php [hunk 2]
```diff
@@ -22,13 +22,16 @@ public function testAccountDetailsAreReturned(): void
...
             'attributes' => [
-                'id' => $user->id,
-                'admin' => false,
+                'uuid' => $user->uuid,
                 'username' => $user->username,
                 'email' => $user->email,
                 'first_name' => $user->name_first,
                 'last_name' => $user->name_last,
-                'language' => $user->language,
+                'language' => 'en',
```
[the remark this answers is not in the document]
Comment (Boy132) [reply tests/Integration/Api/Client/AccountControllerTest.php:30]: The UserFactory always creates a user with `en` as language.

File: app/Transformers/Api/Client/UserTransformer.php [hunk 3]
```diff
@@ -19,15 +18,19 @@ public function getResourceName(): string
...
-            'image' => 'https://gravatar.com/avatar/' . md5(Str::lower($model->email)),
-            '2fa_enabled' => $model->use_totp,
-            'created_at' => $model->created_at->toAtomString(),
+            'uuid' => $user->uuid,
+            'username' => $user->username,
+            'email' => $user->email,
+            'first_name' => $user->name_first,
+            'last_name' => $user->name_last,
+            'language' => $user->language,
+            'root_admin' => (bool) $user->root_admin,
```
[the remark this answers is not in the document]
Comment (Boy132) [reply app/Transformers/Api/Client/UserTransformer.php:33]: Added `admin` back as deprecated field.

**Review by TekExplorer — COMMENTED** [review]
Was about to comment on the typescript files, but the backwards-compat and 2fa reversion took care of it.

I still maintain it should be called `totp_enabled` but otherwise, its probably good to go.

One possible improvement you could make is to investigate the [Scramble](https://scramble.dedoc.co/) docs to make this auto-documented.

**Review by lancepioch — APPROVED** [review]

**Pull request merged**